crates/ruff_python_formatter/src/other/identifier.rs

```diff
@@ -54,14 +56,16 @@ impl<'a> DotDelimitedIdentifier<'a> {
 
 impl Format<PyFormatContext<'_>> for DotDelimitedIdentifier<'_> {
     fn fmt(&self, f: &mut PyFormatter) -> FormatResult<()> {
-        // An import identifier can contain newlines by inserting continuations (backslashes) after
-        // a dot-delimited segment, as in:
-        // ```python
-        // import foo\
-        //    .bar
-        // ```
-        if memchr::memchr(b'\\', f.context().source()[self.0.range()].as_bytes()).is_some() {
-            text(self.0.as_str(), Some(self.0.start())).fmt(f)
+        // The backslashes are line continuations
+        if f.context().source()[self.0.range()]
+            .chars()
+            .any(|c| is_python_whitespace(c) || matches!(c, '\n' | '\r' | '\\'))
+        {
+            let no_whitespace: String = f.context().source()[self.0.range()]
+                .chars()
+                .filter(|c| !is_python_whitespace(*c) && !matches!(c, '\n' | '\r' | '\\'))
+                .collect();
+            text(&no_whitespace, Some(self.0.start())).fmt(f)
```

Comment:
Annoying that we pass `&str` in here only for it to be converted to `String` internally when formatting, but I don't see any alternative APIs.

Reply:
We could change the text element to use `Cow`, there are more places where we had owned strings previously.